Engine/HistoricalData/SubscriptionDataReaderHistoryProvider.cs

```diff
@@ -79,6 +79,11 @@ public override IEnumerable<Slice> GetHistory(IEnumerable<HistoryRequest> reques
         /// </summary>
         protected virtual Subscription CreateSubscription(HistoryRequest request, DateTime start, DateTime end)
         {
+            // Tradable dates are defined with the data time zone to access the right source
+            var tradableDates = Time.EachTradeableDay(request.ExchangeHours,
+                start.ConvertFromUtc(request.DataTimeZone),
+                end.ConvertFromUtc(request.DataTimeZone));
+
             // data reader expects these values in local times
             start = start.ConvertFromUtc(request.ExchangeHours.TimeZone);
```

Comment:
I'd be tempted to create a new variable for start (e.g. exchangeTimeStart) rather than re-using the existing one. This is for 2 reasons - firstly, it gives context to the timezone you're expecting and secondly, it prevents errors such as if these two lines were put above the tradableDates variable creation above.

Reply:
Thank you, @adam-may. I used `startTimeLocal` and `endTimeLocal` to be coherent with naming across Lean where "Local" means "Exchange".